src/nvim/message.c

```diff
@@ -2129,8 +2129,8 @@ void msg_puts_len(const char *const str, const ptrdiff_t len, int hl_id, bool hi
   // If redirection is on, also write to the redirection file.
   redir_write(str, len);
 
-  // Don't print anything when using ":silent cmd".
-  if (msg_silent != 0) {
+  // Don't print anything when using ":silent cmd" or empty message.
+  if (msg_silent != 0 || *str == NUL) {
```

Comment:
Without this the change in `msg_puts_display()` resulted in strlen - 1 overflow; there is no need to try to put an empty message onto the screen, or emit it.